Lib9c/Action/RetrieveAvatarAssets.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Libplanet.Crypto;
+using Nekoyume.Model.State;
+using Nekoyume.Module;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType(TypeIdentifier)]
+    public class RetrieveAvatarAssets: ActionBase
+    {
+        public const string TypeIdentifier = "retrieve_avatar_assets";
+        public Address AvatarAddress;
+
+        public RetrieveAvatarAssets()
+        {
+        }
+
+        public RetrieveAvatarAssets(Address avatarAddress)
+        {
+            AvatarAddress = avatarAddress;
+        }
+
+        public override IValue PlainValue => Dictionary.Empty
+            .Add("type_id", TypeIdentifier)
+            .Add("values", Dictionary.Empty.Add("a", AvatarAddress.Serialize()));
+
+        public override void LoadPlainValue(IValue plainValue)
+        {
+            var asDict = (Dictionary)((Dictionary)plainValue)["values"];
+            AvatarAddress = asDict["a"].ToAddress();
+        }
+
+        public override IWorld Execute(IActionContext context)
+        {
+            context.UseGas(1);
+            Address signer = context.Signer;
+            var state = context.PreviousState;
+            if (state.TryGetAvatarState(signer, AvatarAddress, out _))
+            {
+                var currency = state.GetGoldCurrency();
+                var balance = state.GetBalance(AvatarAddress, currency);
+                if (balance > 0 * currency)
+                {
+                    return state.TransferAsset(context, AvatarAddress, signer, balance);
+                }
```

Comment:
If there is no enough balance (i.e., no NCG), it should throw an exception like `Insufficient...Exception`. If not, it will throw `FailedLoadStateException` and it makes workers on debugging this action confused.

Reply:
If the balance is insufficient, it will be throw InsufficientBalanceException internally in TransferAsset interface, so there is no need to implement it separately. i'll check and add a test case.